client/platform/desktop/frontend/store/dataset.ts

```diff
@@ -45,29 +46,41 @@ const datasets = ref({} as Record<string, JsonMetaCache>);
 
 const recents = computed(() => (Object.values(datasets.value)));
 
+async function autoDiscover() {
+  datasets.value = {};
+  /* Nothing came from localStorage, try to populate from autodiscovery */
+  const discovered: JsonMeta[] = await ipcRenderer.invoke('autodiscover-data');
+  discovered.forEach((d) => {
+    Vue.set(datasets.value, d.id, hydrateJsonMetaCacheValue(d));
+  });
+}
```

Comment:
At the end of this function you may want to populate the localStorage with the values generated.  localStorage is written to during the loading of a dataset (so actually opening a dataset).  You can get into a loop of loading the system with a blank localStorage, go to Settings to sync, then closing.  If you open it again the localStorage wasn't written so it needs to populate it again.  This is fine if you fix the above error, just that it will be continually syncing on each launch until a user launches a dataset.